MobileSdk/src/main/java/com/spruceid/mobile/sdk/Credential.kt

```diff
@@ -32,14 +33,21 @@ private fun Mdoc.jsonEncodedDetailsInternal(elementIdentifiers: List<String>?):
                 }
 
                 if (jsonString != null) {
-                    val json: JSONObject
+                    val jsonObject: JSONObject
                     try {
-                        json = JSONObject(jsonString)
-                    } catch (e: Error) {
+                        jsonObject = JSONObject(jsonString)
+                        return@map Pair(id, jsonObject)
+                    } catch (e: Exception) {
                         print("failed to decode '$id' as JSON: $e")
-                        return@map null
                     }
-                    return@map Pair(id, json)
+
+                    try {
+                        val jsonArray = JSONArray(jsonString)
+                        return@map Pair(id, jsonArray)
+                    } catch (e: Exception) {
+                        print("failed to decode '$id' as JSON: $e")
+                        return@map Pair(id, jsonString)
+                    }
```

Comment:
Ah this is a really good point, most of the time the values will be JSON Strings and Integers, rather than Arrays or Objects. I'll look into how to parse this properly.